pysat/instruments/methods/nasa_cdaweb.py

```diff
@@ -201,8 +202,9 @@ def download(supported_tags, date_array, tag, sat_id,
                                         formatted_remote_fname))
                 req = requests.get(remote_path)
                 if req.status_code != 404:
-                    open(saved_local_fname, 'wb').write(req.content)
-                    logger.info('Finished.')
+                    with TemporaryFileLock(saved_local_fname + '.Lock', pysat.file_timeout) as tfl:
```

Comment:
Any ideas why Asher is using, what looks to be, a lock on a temporary file to mediate writing to the true file? Why note have the lock on the true file directly?